src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs

```diff
@@ -107,7 +107,8 @@ public static void XmlBaseWriter_CheckAsync_ThrowInvalidOperationException()
             var t2 = Assert.ThrowsAsync<InvalidOperationException>(() => writer.WriteBase64Async(bytes, 0, byteSize));
 
             InvalidOperationException e = t2.Result;
-            Assert.StrictEqual(e.Message, "An asynchronous operation is already in progress.");
+            bool isAsyncIsRunningException = e.Message.Contains("XmlAsyncIsRunningException") || e.Message.Contains("in progress");
```

Comment:
>What if the exception message is localized? Should we even be testing the content of the message against a particular string?

We do test exception messages at times...but more to validate the string-id and not the actual text.  Our System.Net.WebSockets.Client tests do this because we want to make sure that we are throwing the correct exception and error message.

I would recommend, though, that we don't do the string compare directly with literal strings in the tests. Instead, the strings.resx file should be used to do the comparison. This avoids hard-coding strings in the test sources themselves.